src/pybind/mgr/cephadm/module.py

```diff
@@ -2198,8 +2198,14 @@ def describe_service(self, service_type: Optional[str] = None, service_name: Opt
             if service_name is not None and service_name != nm:
                 continue
 
-            if spec.service_type != 'osd':
+            if spec.service_type not in ['osd', 'node-proxy']:
```

Comment:
I suggest re-arranging this logic a little. Currently the `else` statement handles osds. And this if clause handles everything else. it would be more logical for the flow to go something like:
```
if spec.service_type == 'osd':
   # do osd stuff
   ...
elif spec.service_type == 'node-proxy':
   # do node proxy stuff
   ...
else:
   # do non-specialized stuff
   ...
```

Future readers will thank you ;-)

Reply:
I didn't realize that this was a backport when I wrote the above. I just saw the reef backport of this patch and then realized the situation. Because this is a backport I will retract my request for changes but my general comment stands and the next time somebody touches this code I hope the if-blocks get rearranged in a more logical order.

Reply:
https://github.com/ceph/ceph/pull/56933